base/lock.jl

```diff
@@ -51,6 +51,19 @@ end
 
 assert_havelock(l::ReentrantLock) = assert_havelock(l, l.locked_by)
 
+show(io::IO, ::ReentrantLock) = print(io, ReentrantLock, "()")
```

Comment:
maybe we should print a unique identifier for each? (also for Condition)
```suggestion
show(io::IO, l::ReentrantLock) = print(io, ReentrantLock, "(#", repr(objectid(l)), ")")
```

Reply:
do you mean:
```suggestion
show(io::IO, l::ReentrantLock) = print(io, ReentrantLock, "(#=", repr(objectid(l)), "=#)") 
```
?

Reply:
Not necessarily. Why does it need to be equals?

Reply:
I thought it was an inline comment like https://github.com/JuliaLang/julia/pull/55239#issuecomment-2250090083

Reply:
I think a single `#` is weird since it will format the rest of the line as a comment:
```julia
julia> l
ReentrantLock(#0x553b23787bd58da3) (locked by Task (runnable) @0x000000010a2a0010)
```